src/main/java/io/strimzi/ValidatingWebhook.java

```diff
@@ -108,41 +118,39 @@ public AdmissionReview webhook(AdmissionReview review) {
                 .build();
     }
 
-    void annotatePodForRestart(String name, String namespace, boolean dryRun)    {
-        Pod pod = client.pods().inNamespace(namespace).withName(name).get();
-
-        if (pod != null) {
-            if (pod.getMetadata() != null
-                    && pod.getMetadata().getLabels() != null
-                    && "Kafka".equals(pod.getMetadata().getLabels().get("strimzi.io/kind"))) {
-                if (pod.getMetadata().getAnnotations() == null) {
-                    pod.getMetadata().setAnnotations(Map.of("strimzi.io/manual-rolling-update", "true"));
-                    LOG.info("Pod {} in namespace {} should be annotated for restart", name, namespace);
-                    patchPod(name, namespace, pod, dryRun);
-                } else if (!"true".equals(pod.getMetadata().getAnnotations().get("strimzi.io/manual-rolling-update"))) {
-                    Map<String, String> newAnnos = new HashMap<>(pod.getMetadata().getAnnotations());
-                    newAnnos.put("strimzi.io/manual-rolling-update", "true");
-                    pod.getMetadata().setAnnotations(newAnnos);
-                    LOG.info("Pod {} in namespace {} should be annotated for restart", name, namespace);
-                    patchPod(name, namespace, pod, dryRun);
-                } else {
-                    LOG.info("Pod {} in namespace {} is already annotated for restart", name, namespace);
-                }
-
+    void annotatePodForRestart(Pod pod, boolean dryRun) {
+        String name = pod.getMetadata().getName();
+        String namespace = pod.getMetadata().getNamespace();
+        if (pod.getMetadata() != null
+                && pod.getMetadata().getLabels() != null
+                && "Kafka".equals(pod.getMetadata().getLabels().get("strimzi.io/kind"))) {
```

Comment:
Would it make sense to move this check to `matchingLabel` method? You would have all checks in one place. It made sense previously here as we did not get the pod earleir. But now it would make sense to move it. Sorry for not noticing it earlier.